### ISSUE TITLE
How to config to only cache on http statuscode 2xx?

### ISSUE THREAD

**Issue by gebrits** [jpodwys/superagent-cache#11]
How would I configure superagent-cache to only cache on http statuscode 2xx?
Is this supported?

**Comment by jpodwys** [comment]
You could make this work by combining the `.cacheWhenEmpty()` and `.prune()` chainable functions as shown below. This is, however, a rather verbose solution. I'm open to adding shorthand support for caching based on response code by implementing some built-in `.prune` selections or by providing a more generic `.onlyCacheWhen()` chainable or something. Let me know what you think.

```javascript
var prune = function(r){
  if(r && r.statusCode && r.statusCode.toString()[0] === '2'){
    //Do not simply return r here or you will be caching the entire response
    //See the docs for .prune() if you need more information
    return r.someProperty;
  }
  return null;
}

superagent
  .get('url')
  .cacheWhenEmpty(false)
  .prune(prune)
  .end(function (err, response) {
    //Your code goes here
  }
);
```

**Comment by jpodwys** [comment]
I just published `1.1.0` to NPM. It adds global configurability to superagent-cache. This means that you could do what you want without adding `.cacheWhenEmpty(false)` and `.prune(prune)` to every query you make. Here's how it would look for you:

```javascript
var superagent = require('superagent');
var defaults = {
  cacheWhenEmpty: false,
  prune: function(r){
    if(r && r.statusCode && r.statusCode.toString()[0] === '2'){
      return r.body;
    }
    return null;
  }
};
require('superagent-cache')(superagent, null, defaults);
```

Then your queries could just look like this while still enjoying your defaults:

```javascript
superagent
  .get('url')
  .end(function (err, response) {
    //Your code goes here
  }
);
```

Does this info help?
